src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionUserCancelHandler.java

```diff
@@ -48,6 +50,13 @@ public Mono<BaseTransactionEvent<?>> handle(TransactionUserCancelCommand command
         return transaction
                 .filter(tx -> tx.getStatus().equals(TransactionStatusDto.ACTIVATED))
                 .switchIfEmpty(Mono.error(new AlreadyProcessedException(command.getData())))
+                .cast(TransactionActivated.class)
+                .filter(
+                        tx -> ((command.getXUserId() == null && tx.getTransactionActivatedData().getUserId() == null) ||
+                                (command.getXUserId() != null && command.getXUserId().toString()
+                                        .equals(tx.getTransactionActivatedData().getUserId())))
+                )
```

Comment:
```suggestion
                .filter(Objects.equals(Objects.toString(command.getXUserId())),tx.getTransactionActivatedData().getUserId())
```

this condition can be simplified using java.utils.Objects.equals method that return true if both input objects are null or equals be meaning of equals method

Reply:
Sounds good, but the `command.getXUserId()` could be null, and applying this code i have an error since `Object.equals `is static and cannot be used in non-static context

Reply:
Ok I see that if the `command.getXUserId()` is null it's not a problem